treeherder/perf/management/commands/reassign_perf_data.py

```diff
@@ -24,19 +29,92 @@ def add_arguments(self, parser):
             '--to',
             action='append',
             help='New signature we want to move performance data to '
-                 '(specify multiple times to get multiple signature)'
+                 '(specify multiple times to get multiple signatures)'
+        )
+        parser.add_argument(
+            '--for',
+            action='store',
+            choices=USE_CASES,
+            metavar='USE CASE',
+            help='''Rename "old" Raptor tp6 subtests, by pointing perf alerts & datum to new signatures.
+                 Cannot be used in conjunction with --from/--to arguments.
+                 Available use cases: {}'''.format(','.join(USE_CASES))
+        )
+        parser.add_argument(
+            '--keep-leftovers',
+            action='store_true',
+            help='Keep database rows even if they become useless after the script runs'
         )
 
     def handle(self, *args, **options):
         from_signatures = options['from']
         to_signatures = options['to']
+        use_case = options['for']
+        keep_leftovers = options['keep_leftovers']
+
+        self.validate_arguments(from_signatures, to_signatures, use_case)
+
+        if use_case:
+            signature_pairs = self.fetch_signature_pairs(use_case)
+        else:
+            signature_pairs = zip(from_signatures, to_signatures)
+
+        with transaction.atomic():
+            self.reassign(signature_pairs)
+            if not keep_leftovers:
+                old_signatures = [old_signature for old_signature, _ in signature_pairs]
+                self.remove_signatures(old_signatures)
+
+    def validate_arguments(self, from_signatures, to_signatures, perf_framework):
+        arguments = [from_signatures, to_signatures, perf_framework]
+        if all(arguments) or not any(arguments):
+            raise CommandError("Use either --for or --from/--to combo")
+
+        if not perf_framework:
+            # --from/--to combo
+            if len(from_signatures) != len(to_signatures):
+                raise CommandError("Each old signature must have a corresponding new one")
+
+    def fetch_signature_pairs(self, use_case):
+        if use_case == RAPTOR_TP6_SUBTESTS:
+            return self.fetch_tp6_signature_pairs()
+
+    def fetch_tp6_signature_pairs(self):
+        with connection.cursor() as cursor:
+            cursor.execute("""
+            SELECT
+                old_signature.id AS old_id,
+                new_signature.id AS new_id
+            FROM
+                performance_signature AS old_signature,
+                performance_signature AS new_signature
+            WHERE
+               old_signature.framework_id = 10 AND
+               old_signature.suite LIKE 'raptor-tp6%' AND
+               old_signature.test LIKE 'raptor-tp6%' AND
+               old_signature.parent_signature_id IS NOT NULL AND
+               old_signature.repository_id IN (1, 2, 6, 77) AND
```

Comment:
can we define what these repository id's are- it will help with debugging in the future. ideally this is defined in a variable with comments.